flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/operations/SourceQueryOperation.java

```diff
@@ -39,9 +39,16 @@
 public class SourceQueryOperation implements QueryOperation {
 
     private final ContextResolvedTable contextResolvedTable;
+    private final Map<String, String> dynamicOptions;
 
     public SourceQueryOperation(ContextResolvedTable contextResolvedTable) {
+        this(contextResolvedTable, Collections.emptyMap());
+    }
+
+    public SourceQueryOperation(
+            ContextResolvedTable contextResolvedTable, Map<String, String> dynamicOptions) {
```

Comment:
I  feel it is better to use Nullable. `scan(t)` and `scan(t, dynamicOptions)` are different.